### PR TITLE
Limit documenting special members to __call__

### LINKED ISSUE #20080: API docs currently include entries for class `__dict__`, `__module__`, `__weakref__`
### Problem

The attributes mentioned in the issue title are clearly internal and should not be documented (*possibly* unless we explicitly override them, if ever).

See e.g. https://matplotlib.org/stable/api/_as_gen/matplotlib.colors.Colormap.html#matplotlib.colors.Colormap.__dict__ https://matplotlib.org/stable/api/_as_gen/matplotlib.colors.Colormap.html#matplotlib.colors.Colormap.__module__ https://matplotlib.org/stable/api/_as_gen/matplotlib.colors.Colormap.html#matplotlib.colors.Colormap.__weakref__

### Suggested Improvement

Suppress these attributes in the docs.

**Matplotlib version**
  * Matplotlib documentation version (is listed under the logo): both 3.4.1 and 3.4.1.post544.

### PR DESCRIPTION
**Pull request by timhoffm**
Let's try and see if this works.

Closes #20080.

Follow up to #17151.

### CODE REVIEW

**Comment by QuLogic** [comment]
Fails with only one warning:
> doc/devel/documenting_mpl.rst:709: WARNING: py:obj reference target not found: matplotlib.patches.Patch.__init__

File: doc/_templates/autosummary.rst [hunk 1]
```diff
@@ -7,7 +7,7 @@
 {% if objtype in ['class'] %}
 .. auto{{ objtype }}:: {{ objname }}
     :show-inheritance:
-    :special-members:
+    :special-members: __call__
```
Comment (tacaswell) [inline doc/_templates/autosummary.rst:10]:
```suggestion
    :special-members: __call__,__init__
```

Comment (timhoffm) [reply doc/_templates/autosummary.rst]: This is not great. We don't want all the `__init__` in the docs (https://57666-1385122-gh.circle-artifacts.com/0/doc/build/html/search.html?q=__init__). It's information is already available in the class docstring, e.g.

grafik

File: doc/devel/documenting_mpl.rst [hunk 2]
```diff
@@ -707,7 +707,7 @@ Then in any function accepting `~.Line2D` pass-through ``kwargs``, e.g.,
       """
 
 Note there is a problem for `~matplotlib.artist.Artist` ``__init__`` methods,
-e.g., `matplotlib.patches.Patch.__init__`, which supports ``Patch`` ``kwargs``,
```
Comment (timhoffm) [inline doc/devel/documenting_mpl.rst:710]: We can't link `__init__` as this is not an explicit entry in the docs. It's contents is folded into the class docstring.

**Review by jklymak — APPROVED** [review]

**Review by QuLogic — APPROVED** [review]

**Pull request merged**